Remove redundant etcd config info

djosborne opened the pull request:
Fixes #218

- Remove duplicate calicoctl config information from 2nd etcd doc
- Remove RBAC walkthrough. Replace with reference to official coreos/etcd docs + calico's bare minimum calico RBAC requirements
- Remove lots of other redundant info, including:
    - half-baked SSL generation guide
    - Explicit warnings about sudo user environment variable inheritance
    - explicit mentions of how to configure docker daemon with certs

Future work, still need:
- calico-node etcd configuration guide.
- felix etcd configuration guide

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
```diff
@@ -1,132 +1,63 @@
 The file location may be overridden using the `--config` option on commands that required
 datastore access.
 
-The config file is a yaml map, with the following supported fields: 
```
Not sure about the changes to this file... I don't think we should be getting rid of the reference table here + the example commands.  I think this was mostly fine as it was.

caseydavenport commented on master/reference/advanced/etcd-secure.md:
```diff
@@ -1,84 +0,0 @@
----
```
Yep, I think we want to get rid of this doc.

caseydavenport commented on master/reference/advanced/secure-etcd.md:
```diff
@@ -0,0 +1,98 @@
+---
+title: Securing etcd
```
"Securing etcd" feels kinda generic / not quite right as a title.

I'm also not convinced that "advanced" is the right place for this doc.

I think there are two parts:

- Reference guide for configuring access to etcd for various components.
  - We've already got one for calicoctl (reference/calicoctl/setup/...)
  - We've already got one for CNI (though not in this repo)
  - Not sure we've got one for calico/node.  If not, then we should have a calico/node reference guide for all config options (including etcd stuff).
- A user tutorial for etcd RBAC (which should just be a link to etcd docs, if possible)

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
```diff
@@ -1,132 +1,63 @@
+title: Connecting calicoctl to etcd
 ---
 
-This document covers the configuration options for calicoctl when using an etcdv2 datastore.
```
Why did you get rid of this line?

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
```diff
@@ -1,132 +1,63 @@
-
-# Additional information
-
-## Configure role based access for etcd v2
```
Yeah, as I said a few places above, I think RBAC should be moved to another guide that's more of a "user walkthrough / tutorial", or just removed and included as a link to the etcd docs.

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
````diff
@@ -1,132 +1,63 @@
-```
-$ etcdctl user grant calicoctl -roles calico-readwrite
-```
+#### Root Environment Variables
````
We've already got a FAQ on this - I'd rather not duplicate the info.

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
```diff
@@ -1,132 +1,63 @@
-`etcdEndpoints` can also be represented as `ETCD_ENDPOINTS`.
-
-#### Example configuration file
+The following is an example `calicoctl.cfg`, which lists all supported fields:
```
I don't really like this approach to documenting our configuration options.  Examples are not reference documentation.  They can't express all the stuff we might want to express - we should be using tables which outline the valid configuration options rather than examples.

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
````diff
@@ -1,132 +1,63 @@
-#### Example using environment variables 
-
-```
-ETCD_ENDPOINTS=http://myhost1:2379 calicoctl get bgppeers
````
I think we want to keep this example.

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
````diff
@@ -1,132 +1,63 @@
-```
-$ etcdctl -u root:<rootpw> auth disable
-```
+## calico/node
````
I think this belongs in the `calico/node` doc (which may not exist yet).

This document is about configuring calicoctl to interact with an etcdv2 cluster.

djosborne commented:
@caseydavenport back to you. I reworked this significantly. Trying to change too much at once. Updated the PR description to reflect what this addresses. Hoping to get this merged as an increment as our etcd docs are so spread out and cover so much its easy to tackle too mnuch at once

caseydavenport commented on master/reference/advanced/etcd-rbac.md:
```diff
@@ -0,0 +1,29 @@
+component configuration guide:
+
+- [calicoctl]({{site.baseurl}}/{{page.version}}/reference/calicoctl/setup/etcdv2)
+- [calico-cni]({{site.baseurl}}/{{page.version}}/reference/cni-plugin/configuration)
```
Does the CNI plugin actually support RBAC?

caseydavenport commented on master/usage/configuration/as-service.md:
```diff
@@ -9,8 +9,8 @@ daemons such as upstart as well.
 
 ## Running the Calico Node Container as a Service
```
While we're at it, can we make this title better? A "Service" is a pretty vague thing.

This document is really "Running calico/node using systemd"

djosborne replied:
ah god. IMO this doc needs a rewrite in conjunction with when we write the "calico-node etcd configuration guide.". I'd rather not touch it beyond what I have to right now.

caseydavenport commented on master/usage/configuration/as-service.md:
```diff
@@ -9,8 +9,8 @@ daemons such as upstart as well.
 This section describes how to run the Calico node as a Docker container
-in Systemd.  Included here is an EnvironmentFile that defines the Environment 
-variables for Calico and a sample systemd service file that uses the 
+in Systemd.  Included here is an EnvironmentFile that defines the Environment
```
"systemd" shouldn't be capitalized

caseydavenport commented on master/reference/calicoctl/setup/etcdv2.md:
```diff
@@ -1,14 +1,14 @@
 ---
-title: Configuring calicoctl - etcdv2 datastore 
-layout: docwithnav
+title: Using calicoctl With an etcdv2 Datastore
```
What about the equivalent title for the Kubernetes backend?  They should have parallel structure, so if you're changing the title of this one make sure you're also doing it there :)

caseydavenport commented:
```
- /_site/master/usage/configuration/securing-calico.html
  *  internally linking to /master/reference/advanced/etcd-secure, which does not exist (line 371)
     <a href="/master/reference/advanced/etcd-secure">calicoctl configuration
```

Build failure and a few comments, but lookin' good.

caseydavenport commented:
LGTM.
